send_message_to_msteams.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import random
+from abc import ABC, abstractmethod
+
+import pymsteams
+import requests
+from requests.exceptions import ConnectionError, HTTPError
+
+logging.basicConfig(
+    level=logging.INFO, 
+    format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+TEAMS_WEBHOOK_URL = "your_webhook"  # TODO: store credentials in a secure way
+
+
+class ContentGenerator(ABC):
+    @abstractmethod
+    def get_content(self):
+        """ Method create connection to API and select random image/quote/etc."""
+        pass
+
+
+class QuoteGeneratorFromQuotableAPI(ContentGenerator):
+    QUOTABLE_API_URL = "https://api.quotable.io/random"
+
+    def get_content(self):
+        response = requests.get(self.QUOTABLE_API_URL)
+
+        if response.status_code != 200:
+            logging.error("Failed to retrieve a quote")
+            return None
+        else:
+            data = response.json()
+            quote_text = data["content"]
+            quote_author = data["author"]
+            logging.info("A quote successfully generated.")
+
+            return quote_text, quote_author
+
+
+class ImageGeneratorFromPiscumAPI(ContentGenerator):
+    WIDTH = random.randint(300, 800)
+    HEIGHT = random.randint(300, 800)
+    PISCUM_API_URL = "https://picsum.photos"
+
+    def get_content(self):
+        image_url = f"{self.PISCUM_API_URL}/{self.WIDTH}/{self.HEIGHT}"
+        return image_url
+
+
+class TeamsMessageSender:
+    MESSAGE_COLOR = "#F8C471"
+    
+    def __init__(self, webhook_url):
+        self.webhook_url = webhook_url
+
+    def send_message(self, title, quote_text, quote_author, image_url, max_attempts=3, delay=5):
+        """Method combines the whole message and send it to MSTeams channel."""
+        attempt = 0
+        myTeamsMessage = pymsteams.connectorcard(self.webhook_url)
+        myTeamsMessage.color(self.MESSAGE_COLOR)
+        myTeamsMessage.title(title)
+        myTeamsMessage.text(
+            f"**Quote:** {quote_text}\n\n" 
+            f"**Author:** {quote_author}\n\n"
+            f"![Image]({image_url})"
+        )
+        while attempt < max_retries:
```

Comment:
In Airflow we have a separate retry mechanism, better to remove it 

BTW, in python it's also better to implement it with a decorator - if you need more details, refer to me in private messages